Does not compile on rustc 1.15.0-nightly (ac635aa95 2016-11-18)

```
$ cargo build
   Compiling clippy_lints v0.0.99
error[E0308]: mismatched types
  --> clippy_lints/src/len_zero.rs:71:86
   |
71 |             ItemImpl(_, _, _, None, _, ref impl_items) => check_impl_items(cx, item, impl_items),
   |                                                                                      ^^^^^^^^^^ expected slice, found struct `syntax::ptr::P`
   |
   = note: expected type `&[rustc::hir::ImplItem]`
   = note:    found type `&syntax::ptr::P<[rustc::hir::ImplItemRef]>`

error: aborting due to previous error

error: Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.

$ rustc --version
rustc 1.15.0-nightly (ac635aa95 2016-11-18)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,9 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## 0.0.100 — 2016-11-20
+* Update to *rustc 1.15.0-nightly (ac635aa95 2016-11-18)*
+
 ## 0.0.99 — 2016-11-18
 * Update to rustc 1.15.0-nightly (0ed951993 2016-11-14)
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "clippy"
-version = "0.0.99"
+version = "0.0.100"
 authors = [
 	"Manish Goregaokar <manishsmail@gmail.com>",
 	"Andre Bogus <bogusandre@gmail.com>",
@@ -25,7 +25,7 @@ test = false
 
 [dependencies]
 # begin automatic update
-clippy_lints = { version = "0.0.99", path = "clippy_lints" }
+clippy_lints = { version = "0.0.100", path = "clippy_lints" }
 # end automatic update
 
 [dev-dependencies]

--- a/clippy_lints/Cargo.toml
+++ b/clippy_lints/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "clippy_lints"
 # begin automatic update
-version = "0.0.99"
+version = "0.0.100"
 # end automatic update
 authors = [
 	"Manish Goregaokar <manishsmail@gmail.com>",

--- a/clippy_lints/src/len_zero.rs
+++ b/clippy_lints/src/len_zero.rs
@@ -92,7 +92,11 @@ fn check_trait_items(cx: &LateContext, item: &Item, trait_items: &[TraitItem]) {
     fn is_named_self(item: &TraitItem, name: &str) -> bool {
         item.name.as_str() == name &&
         if let MethodTraitItem(ref sig, _) = item.node {
-            is_self_sig(sig)
+            if sig.decl.has_self() {
+                sig.decl.inputs.len() == 1
+            } else {
+                false
+            }
         } else {
             false
         }
@@ -111,18 +115,22 @@ fn check_trait_items(cx: &LateContext, item: &Item, trait_items: &[TraitItem]) {
     }
 }
 
-fn check_impl_items(cx: &LateContext, item: &Item, impl_items: &[ImplItem]) {
-    fn is_named_self(item: &ImplItem, name: &str) -> bool {
+fn check_impl_items(cx: &LateContext, item: &Item, impl_items: &[ImplItemRef]) {
+    fn is_named_self(cx: &LateContext, item: &ImplItemRef, name: &str) -> bool {
         item.name.as_str() == name &&
-        if let ImplItemKind::Method(ref sig, _) = item.node {
-            is_self_sig(sig)
+        if let AssociatedItemKind::Method { has_self } = item.kind {
+            has_self && {
+                let did = cx.tcx.map.local_def_id(item.id.node_id);
+                let impl_ty = cx.tcx.item_type(did);
+                impl_ty.fn_args().skip_binder().len() == 1
+            }
         } else {
             false
         }
     }
 
-    let is_empty = if let Some(is_empty) = impl_items.iter().find(|i| is_named_self(i, "is_empty")) {
-        if cx.access_levels.is_exported(is_empty.id) {
+    let is_empty = if let Some(is_empty) = impl_items.iter().find(|i| is_named_self(cx, i, "is_empty")) {
+        if cx.access_levels.is_exported(is_empty.id.node_id) {
             return;
         } else {
             "a private"
@@ -131,8 +139,8 @@ fn check_impl_items(cx: &LateContext, item: &Item, impl_items: &[ImplItem]) {
         "no corresponding"
     };
 
-    if let Some(i) = impl_items.iter().find(|i| is_named_self(i, "len")) {
-        if cx.access_levels.is_exported(i.id) {
+    if let Some(i) = impl_items.iter().find(|i| is_named_self(cx, i, "len")) {
+        if cx.access_levels.is_exported(i.id.node_id) {
             let def_id = cx.tcx.map.local_def_id(item.id);
             let ty = cx.tcx.item_type(def_id);
 
@@ -143,14 +151,6 @@ fn check_impl_items(cx: &LateContext, item: &Item, impl_items: &[ImplItem]) {
                                ty,
                                is_empty));
         }
-    }
-}
-
-fn is_self_sig(sig: &MethodSig) -> bool {
-    if sig.decl.has_self() {
-        sig.decl.inputs.len() == 1
-    } else {
-        false
     }
 }
 

